### ISSUE TITLE
Ruby 3.0 test failure

### ISSUE THREAD

**Issue by voxik** [eventmachine/eventmachine#941]
Running test suite against [Ruby 3.0](https://github.com/eventmachine/eventmachine/pull/923/checks?check_run_id=1887191394#step:7:381), there is one failure:

~~~
1) Failure: test_em_system_pid(TestProcesses)
/home/runner/work/eventmachine/eventmachine/tests/test_processes.rb:51:in `test_em_system_pid'
     48:         $pids << EM.system('echo hi', proc{ |out,status|$pids << status.pid; EM.stop })
     49:       }
     50: 
  => 51:       assert_equal $pids[0], $pids[1]
     52:     end
     53: 
     54:     def test_em_system_with_proc
<1987> expected but was
<0>
~~~

Just trying to break in the  proc, it seems that the `status` variable contains uninitialized `Process::Status` object:

~~~
irb(EventMachine):002:0> 

From: /builddir/build/BUILD/eventmachine-1.2.7/usr/share/gems/gems/eventmachine-1.2.7/tests/test_processes.rb @ line 48 :

    43: 
    44:     def test_em_system_pid
    46: 
    47:       EM.run{
 => 48:         $pids << EM.system('echo hi', proc{ |out,status|binding.irb; $pids << status.pid; EM.stop })
    49:       }
    50: 
    51:       assert_equal $pids[0], $pids[1]
    52:     end
    53: 

irb(test_em_system_pid(TestProcesses)):001:0> p out
"hi\n"
=> "hi\n"
irb(test_em_system_pid(TestProcesses)):002:0> p status
#<Process::Status: uninitialized>
=> #<Process::Status: uninitialized>
irb(test_em_system_pid(TestProcesses)):003:0> p status.pid
0
=> 0
~~~

**Comment by MSP-Greg** [comment]
@voxik See PR #948, seems to fix the issue...

**Comment by sodabrew** [comment]
Resolved by #948
